samples/crypto/test_sample/README.rst

```diff
@@ -0,0 +1,258 @@
+The ``p_test_vector_name`` variable is the only required member in the test_vector structures::
+
+   typedef const struct
+   {
+       const nrf_crypto_hash_info_t    * p_hash_info;                  /**< Pointer to hash info type. */
+       const ret_code_t                  expected_err_code;            /**< Expected error code from hash operation. */
+       const uint8_t                     expected_result;              /**< Expected result of hash operation. */
+       const hash_mem_mode_t             mode;                         /**< Hash memory operation. */
+       const uint32_t                    chunk_length;                 /**< Size of input chunks to hash function in bytes. */
+       const uint32_t                    update_iterations;            /**< Number of update iterations of input. */
+       const char                      * p_test_vector_name;           /**< Pointer to hash test vector name. */
+       const char                      * p_input;                      /**< Pointer to input message in hex string format. */
+       const char                      * p_expected_output;            /**< Pointer to expected message digest in hex string format. */
+   }test_vector_hash_t;
+
+Registering a test vector
+-------------------------
+
+Test vectors are added by registering them for a section defined in the test case code.
+The test vector is registered in the section ``test_vector_hash_data`` which is defined in the test case example ``exec_test_case_hash``.
+The test vector can reuse the already defined hash test vector structure ``test_vector_hash_t``, as shown in the code block below::
+
+   // Register a test vector for sha256.
+   NRF_SECTION_ITEM_REGISTER(test_vector_hash_data, test_vector_hash_t test_vector_sha256) =
+   {
+       .p_hash_info        = &g_nrf_crypto_hash_sha256_info,                                       /**< Pointer to sha256 info type. */
+       .expected_err_code  = NRF_SUCCESS,                                                          /**< The expected error code from the sha operation. */
+       .expected_result    = EXPECTED_TO_PASS,                                                     /**< The expected result of the operation. */
+       .p_test_vector_name = "SHA256 test vector example",                                         /**< Pointer to sha256 test vector name. */
+       .p_input            = "c98c8e55",                                                           /**< Pointer to input message. */
+       .p_expected_output  = "7abc22c0ae5af26ce93dbb94433a0e0b2e119d014f8e7f65bd56c61ccccd9504"    /**< Pointer to expected digest. */
+   };
+
+Output logging
+--------------
+
+The test project generates a test log using RTT transport, as described in the :ref:`crypto_test_testing` section.
+The new test case ``exec_test_case_hash`` and its test vector will be added to the test log when the test is executed.
```

Comment:
Change "The new test case exec_test_case_hash" to "Executing exec_test_case_hmac with its registered test vectors will add..."

Reply:
done